### ISSUE TITLE
Add more dates to X axis?

### ISSUE THREAD

**Issue by TraderMan999** [matplotlib/mplfinance#528]
Is there a way in the api to add more dates on the x axis?  This helps me avoid having to count candles from the candles plotted and the sparsely populated candle dates. 

I’ve been through the docs and may be overlooking it. This would be very helpful. 

Also, does it plot “blue raindrops” like is done on trend spider?

**Comment by DanielGoldfarb** [comment]
> Is there a way in the api to add more dates on the x axis? 

No.  I am presently working on an enhancement that will make this possible.  Hopefully it will be done within a few weeks, however I found that I am being distracted by other projects lately and then have to come back to this enhancement.  Check back in a couple of weeks for status.

> ... plot “blue raindrops” like is done on trend spider?

No.  Will have to look into this.  It appears that "Raindrop Charts" is a [Trademark owned by Trend Spider](https://trendspider.com/product/raindrops/).   I will take a look at the [white paper](https://drt8s3xkrl8yg.cloudfront.net/1/assets/whitepapers/raindrops_280519.pdf) when I have some time.  Between the Trademark name, and the possibility that the algorithm is copyrighted, we will have to see if this can be done without any infringement.

**Comment by TraderMan999** [comment]
Thank you for the response. Can’t wait to see the new feature!!!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ZX2XZX2** [comment]
@TraderMan999: this code will add more dates to the x-axis; it will label every Monday (`df` is the dataframe you want to chart):
```
fig = mpf.figure(figsize=(10, 6), style='yahoo')
ax1 = fig.add_subplot(3, 1, (1, 2))
ax2 = fig.add_subplot(3, 1, 3, sharex=ax1)
fig.subplots_adjust(hspace=0)
xticks, xticklabels = [], []
mth = -1
for i, dt in enumerate(df.index):
    if dt.dayofweek == 0:
        xticks.append(i)
        if dt.month != mth:
            mth = dt.month
            xticklabels.append(datetime.strftime(dt, '%b %d'))
        else:
            xticklabels.append(datetime.strftime(dt, '%d'))
        ax1.set_xticks(xticks)
        ax1.set_xticklabels(xticklabels)
mpf.plot(df, type='candle', ax=ax1, volume=ax2, axtitle=title)
```

**Comment by TraderMan999** [comment]
Thx I’ll give it a try!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
